src/qt/rpcconsole.cpp

```diff
@@ -1192,31 +1193,15 @@ void RPCConsole::updateDetailWidget()
         ui->peerPermissions->setText(permissions.join(" & "));
     }
     ui->peerMappedAS->setText(stats->nodeStats.m_mapped_as != 0 ? QString::number(stats->nodeStats.m_mapped_as) : ts.na);
-
-    // This check fails for example if the lock was busy and
-    // nodeStateStats couldn't be fetched.
-    if (stats->fNodeStateStatsAvailable) {
-        ui->peerServices->setText(GUIUtil::formatServicesStr(stats->nodeStateStats.their_services));
-        // Sync height is init to -1
-        if (stats->nodeStateStats.nSyncHeight > -1) {
-            ui->peerSyncHeight->setText(QString("%1").arg(stats->nodeStateStats.nSyncHeight));
-        } else {
-            ui->peerSyncHeight->setText(ts.unknown);
-        }
-        // Common height is init to -1
-        if (stats->nodeStateStats.nCommonHeight > -1) {
-            ui->peerCommonHeight->setText(QString("%1").arg(stats->nodeStateStats.nCommonHeight));
-        } else {
-            ui->peerCommonHeight->setText(ts.unknown);
-        }
-        ui->peerHeight->setText(QString::number(stats->nodeStateStats.m_starting_height));
-        ui->peerPingWait->setText(GUIUtil::formatPingTime(stats->nodeStateStats.m_ping_wait));
-        ui->peerAddrRelayEnabled->setText(stats->nodeStateStats.m_addr_relay_enabled ? ts.yes : ts.no);
-        ui->peerAddrProcessed->setText(QString::number(stats->nodeStateStats.m_addr_processed));
-        ui->peerAddrRateLimited->setText(QString::number(stats->nodeStateStats.m_addr_rate_limited));
-        ui->peerRelayTxes->setText(stats->nodeStateStats.m_relay_txs ? ts.yes : ts.no);
-    }
-
+    ui->peerServices->setText(GUIUtil::formatServicesStr(stats->nodeStateStats.m_services));
```

Comment:
Concept NACK to displaying incorrect data when data is not available.

Reply:
Done for the GUI
```diff
--- a/src/qt/guiutil.cpp
+++ b/src/qt/guiutil.cpp
@@ -748,7 +748,7 @@ QString formatServicesStr(quint64 mask)
     if (strList.size())
         return strList.join(", ");
     else
-        return QObject::tr("None");
+        return QObject::tr("Unknown");
 }
```
and left (for now) the RPC getpeerinfo behavior as it is currently, i.e. unchanged
```
    "services": "0000000000000000",
    "servicesnames": [
    ],
```